src/cascadia/TerminalApp/ActionArgs.cpp

```diff
@@ -15,3 +15,186 @@
 #include "AdjustFontSizeArgs.g.cpp"
 #include "SplitPaneArgs.g.cpp"
 #include "OpenSettingsArgs.g.cpp"
+
+#include <LibraryResources.h>
+
+namespace winrt::TerminalApp::implementation
+{
+    winrt::hstring NewTerminalArgs::GenerateName() const
+    {
+        std::wstringstream ss;
+
+        if (!_Profile.empty())
+        {
+            ss << fmt::format(L"profile: {}, ", _Profile);
+        }
+        else if (_ProfileIndex)
+        {
+            ss << fmt::format(L"profile index: {}, ", _ProfileIndex.Value());
+        }
+
+        if (!_Commandline.empty())
+        {
+            ss << fmt::format(L"commandline: {}, ", _Commandline);
+        }
+
+        if (!_StartingDirectory.empty())
+        {
+            ss << fmt::format(L"directory: {}, ", _StartingDirectory);
+        }
+
+        if (!_TabTitle.empty())
+        {
+            ss << fmt::format(L"title: {}, ", _TabTitle);
+        }
+        auto s = ss.str();
+        if (s.empty())
+        {
+            return L"";
+        }
+
+        // Chop off the last ", "
+        return winrt::hstring{ s.substr(0, s.size() - 2) };
+    }
+
+    winrt::hstring CopyTextArgs::GenerateName() const
+    {
+        if (_SingleLine)
+        {
+            return RS_(L"CopyTextAsSingleLineCommandKey");
+        }
+        return RS_(L"CopyTextCommandKey");
+    }
+
+    winrt::hstring NewTabArgs::GenerateName() const
+    {
+        winrt::hstring newTerminalArgsStr;
+        if (_TerminalArgs)
+        {
+            newTerminalArgsStr = _TerminalArgs.GenerateName();
+        }
+        if (newTerminalArgsStr.empty())
+        {
+            return RS_(L"NewTabCommandKey");
+        }
+        else
+        {
+            return winrt::hstring{ fmt::format(L"{}, {}", RS_(L"NewTabCommandKey"), newTerminalArgsStr) };
+        }
+        return L"NewTabArgs";

```

Comment:
If you drop the contents of the `else` to the function level, you can drop this return